integration_testing/connection_controller_utils_test.go

```diff
@@ -129,6 +129,21 @@ func waitForCountRemoteNodeIndexer(t *testing.T, node1 *cmd.Node, allCount int,
 	waitForCondition(t, fmt.Sprintf("Waiting for Node (%s) to have appropriate RemoteNodes counts", userAgent), condition)
 }
 
+func waitForCountRemoteNodeIndexerHandshakeCompleted(t *testing.T, node1 *cmd.Node, allCount, validatorCount int,
+	nonValidatorOutboundCount int, nonValidatorInboundCount int) {
+
+	userAgent := node1.Params.UserAgent
+	rnManager := node1.Server.GetConnectionController().GetRemoteNodeManager()
+	condition := func() bool {
+		if true != checkRemoteNodeIndexerCountHandshakeCompleted(rnManager, allCount, validatorCount,
+			nonValidatorOutboundCount, nonValidatorInboundCount) {
+			return false
+		}
+		return true
+	}
```

Comment:
This feels like a roundabout way to write this? 
```suggestion
	condition := func() bool {
		return checkRemoteNodeIndexerCountHandshakeCompleted(rnManager, allCount, validatorCount,
			nonValidatorOutboundCount, nonValidatorInboundCount)
	}
```